server/ml/simple_knn_estimator.go

```diff
@@ -0,0 +1,147 @@
+package ml
+
+import (
+	"container/heap"
+	"errors"
+)
+
+const defaultK = 10
+
+// SimpleKNN struct
+type SimpleKNN struct {
+	params                  map[string]interface{}
+	channelSimilarityMatrix [][]float64
+	activityMatrix          [][]float64
+	userIndexes             map[string]int
+	channelIndexes          map[string]int
+	similarity              funcSimilarity
+	k                       int
+}
+
+// BaseEstimator determines interface for all estimators for user-channel suggestions
+type BaseEstimator interface {
+	SetParams(params map[string]interface{})
+	Predict(userID, channelID string) (float64, error)
+	Fit(activities map[string]map[string]int64)
+}
+
+// NewSimpleKNN returns Simple KNN Estimator
+func NewSimpleKNN(params map[string]interface{}) BaseEstimator {
+	simpleKNN := new(SimpleKNN)
+	simpleKNN.SetParams(params)
+	return simpleKNN
+}
+
+// SetParams sets parameters for KNN estimator
+func (knn *SimpleKNN) SetParams(params map[string]interface{}) {
```

Comment:
Nit: You maybe want to use something like https://github.com/tmrts/go-patterns/blob/master/idiom/functional-options.md here